dali/pipeline/data/buffer.h

```diff
@@ -218,11 +218,17 @@ class DLL_PUBLIC Buffer {
   }
 
   /**
-   * @brief Sets the type of allocation (pinned/non-pinned) for
-   * CPU buffers
+   * @brief Return true if there was data allocation
+   */
+  inline bool has_data_allocation() {
```

Comment:
```suggestion
  inline bool has_data {
```
`allocation` is not that clear when data is _shared_, not allocated by this buffer.